trio/_core/_tests/test_multierror.py

```diff
@@ -555,3 +557,36 @@ def test_apport_excepthook_monkeypatch_interaction():
         ["--- 1 ---", "KeyError", "--- 2 ---", "ValueError"],
         stdout,
     )
+
+
+@pytest.mark.parametrize("protocol", range(0, pickle.HIGHEST_PROTOCOL + 1))
+def test_pickle_multierror(protocol) -> None:
+    # use trio.MultiError to make sure that pickle works through the deprecation layer
+    import trio
+
+    my_except = ZeroDivisionError()
+
+    try:
+        1 / 0
+    except ZeroDivisionError as e:
+        my_except = e
+
+    # MultiError will collapse into different classes depending on the errors
+    for cls, errors in (
+        (ZeroDivisionError, [my_except]),
+        (NonBaseMultiError, [my_except, ValueError()]),
+        (MultiError, [BaseException(), my_except]),
+    ):
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore")
```

Comment:
What specifically is this here for?

Also, can we add a test against strict MultiErrors (from `strict_exception_groups=True`) while we're at it? (it should pass, I think?)

Reply:
For ignoring the deprecation warnings on MultiError. I'll specify it in the simplefilter.
I explicitly wanted to use `trio.MultiError` (instead of the `from .._multierror import MultiError` that the rest of the tests in the file uses) to make sure pickling works through the deprecation layer.

I'm not totally sure what you mean, and all the `strict_exception_groups=True` tests in `trio/_core/_tests/test_run.py` are using MultiError - so maybe it's handled over there? Otherwise feel free to push a commit / open a separate PR with the test you're thinking of, I haven't touched `strict_exception_groups` much at all.

Reply:
Sorry, what I mean is just that maybe we should test that the errors that come with `strict_exception_groups=True` (which are still `MultiErrors`, I think? Just non-collapsed, I believe?) are also pickle-able.

Reply:
Ah, thank you!
It fails due to traceback's not being pickle-able, but if I install and enable https://pypi.org/project/tblib/ it works. Although it might be because I'm running the test slightly different and not related to `strict_exception_groups` that the behaviour differs, idk.
So could leave the test out, add it but skip it if `tblib` isn't installed, add `tblib` to test requirements, or play around more with tests to figure out exactly how/where it differs. But regardless I think it works both with&without `strict_exception_groups`.

Reply:
Makes sense. Didn't realize it would require a dependency, all's good.